lib/fog/google/shared.rb

```diff
@@ -97,22 +102,24 @@ def initialize_google_client(options)
             raise ArgumentError.new("Missing required arguments: google_client_email")
           end
 
-          if ENV["DEBUG"]
-            ::Google::Apis.logger = ::Logger.new(::STDERR)
-            ::Google::Apis.logger.level = ::Logger::DEBUG
-          end
-
           auth = ::Google::Auth::ServiceAccountCredentials.make_creds(
             :json_key_io => StringIO.new(json_key_hash.to_json),
             :scope => options[:google_api_scope_url]
           )
         elsif options[:google_auth]
           auth = options[:google_auth]
         else
-          raise ArgumentError.new(
-            "Missing required arguments: google_json_key_location, "\
-            "google_json_key_string or google_auth"
-          )
+          # fall back to Application Default Credentials before erroring
+          begin
+            auth = ::Google::Auth::DefaultCredentials.from_well_known_path(
+              options[:google_api_scope_url]
+            )
+          rescue StandardError
```

Comment:
Please, don't rescue `StandardError`, only the exception class you really want to capture.

Otherwise this may mask other issues and confuse users by throwing `Application Default Credentials not found; Missing required` on any error.